library/manageiq_user.py

```diff
@@ -0,0 +1,268 @@
+    default: null
+  state:
+    description:
+      - the state of the user
+      - On present, it will create the user if it does not exist or update the
+      user if the associated data is different
+      - On absent, it will delete the user if it exists
+    required: false
+    choices: ['present', 'absent']
+  verify_ssl:
+    description:
+      - whether SSL certificates should be verified for HTTPS requests
+    required: false
+    default: True
+    choices: ['True', 'False']
+  ca_bundle_path:
+    description:
+      - the path to a CA_BUNDLE file or directory with certificates
+    required: false
+    default: null
+'''
+
+EXAMPLES = '''
+# Create a new user in ManageIQ
+  manageiq_user:
+    userid: 'dkorn'
+    username: 'Daniel Korn'
+    password: '******'
+    group: 'EvmGroup-user'
+    email: 'dkorn@redhat.com'
+    state: 'present'
+    miq_url: 'http://localhost:3000'
+    miq_username: 'admin'
+    miq_password: '******'
+    verify_ssl: False
+'''
+
+import os
+from miqclient.api import API as MiqApi
+
+
+class ManageIQUser(object):
+    """ ManageIQ object to execute user management operations in manageiq
+
+    url            - manageiq environment url
+    user           - the username in manageiq
+    password       - the user password in manageiq
+    verify_ssl     - whether SSL certificates should be verified for HTTPS requests
+    ca_bundle_path - the path to a CA_BUNDLE file or directory with certificates
+    """
+
+    def __init__(self, module, url, user, password, verify_ssl, ca_bundle_path):
+        self.module        = module
+        self.api_url       = url + '/api'
+        self.user          = user
+        self.password      = password
+        self.client        = MiqApi(self.api_url, (self.user, self.password), verify_ssl=verify_ssl, ca_bundle_path=ca_bundle_path)
+        self.changed       = False
+
+    def find_group_by_name(self, group_name):
+        """ Searches the group name in ManageIQ.
+
+        Returns:
+            the group id if it exists in manageiq, None otherwise.
+        """
+        groups = self.client.collections.groups
+        return next((group.id for group in groups if group.description == group_name), None)
+
+    def find_user_by_userid(self, userid):
+        """ Searches the userid in ManageIQ.
+
+        Returns:
+            the user's id if it exists in manageiq, None otherwise.
+        """
+        users = self.client.collections.users
+        return next((user.id for user in users if user.userid == userid), None)
+
+    def delete_user(self, userid):
+        """Deletes the user from manageiq.
+
+        Returns:
+            a short message describing the operation executed.
+        """
+        user_id = self.find_user_by_userid(userid)
+        if not user_id:  # user doesn't exist
+            self.module.fail_json(
+                msg="Failed to delete user {userid}: user {userid} does not exist in manageiq".format(userid=userid))
```

Comment:
Should this be a failure given `state: absent`?  I think it should simply return changed=False.